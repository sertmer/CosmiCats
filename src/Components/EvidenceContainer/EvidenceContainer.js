import React from 'react';
import './EvidenceContainer.scss';
import Evidence from '../Evidence/Evidence';

export const EvidenceContainer = () => {
  return (
    <Evidence />
  )
}

export default EvidenceContainer;