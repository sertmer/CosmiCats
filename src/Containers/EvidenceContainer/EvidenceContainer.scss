@import '../../_scss/variables';
@import '../../_scss/mixins';

.evidence-container {
  display: flex;
  flex-flow: row wrap;
  width: 100%;
  overflow: auto;
}

