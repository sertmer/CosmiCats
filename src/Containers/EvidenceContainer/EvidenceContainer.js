import React from 'react';
import './EvidenceContainer.scss';
import Evidence from '../Evidence/Evidence';
import { connect } from 'react-redux';

export const EvidenceContainer = ({evidence}) => {
  const displayEvidence = evidence.map(evi => {
    return <Evidence evidence={evi} key={evi.id} />
  })
  return (
    <section className='evidence-container'>
      {displayEvidence}
    </section>
  )
}

export const mapStateToProps = state => ({
  evidence: state.evidence
})

export default connect(mapStateToProps)(EvidenceContainer);