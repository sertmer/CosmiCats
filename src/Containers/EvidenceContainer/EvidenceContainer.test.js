import React from 'react';
import { shallow } from 'enzyme';
import { EvidenceContainer, mapStateToProps } from './EvidenceContainer'

describe('EvidenceContainer', () => {
  let wrapper
  let mockEvidence

  beforeEach(() => {
    mockEvidence = [{
      img: 'someURL.com',
      id: 'ab1',
      dateCreated: '12-12-12'
    }]

    wrapper = shallow(<EvidenceContainer evidence={mockEvidence}/>)
  })

  describe('EvidenceContainer unit tests', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with evidence', () => {
      const mockState = {
        evidence: mockEvidence,
        extra: 'extra'
      }

      const expected = {
        evidence: mockEvidence
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})