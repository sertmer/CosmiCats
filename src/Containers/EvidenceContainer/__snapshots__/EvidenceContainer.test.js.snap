// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceContainer EvidenceContainer unit tests should match the snapshot 1`] = `
<section
  className="evidence-container"
>
  <Connect(Evidence)
    evidence={
      Object {
        "dateCreated": "12-12-12",
        "id": "ab1",
        "img": "someURL.com",
      }
    }
    key="ab1"
  />
</section>
`;
